torch/distributed/device_mesh.py

```diff
@@ -207,7 +207,7 @@ def __init__(
 
             # private field to pre-generate DeviceMesh's hash
             self._flatten_mesh_list = tuple(self.mesh.flatten().tolist())
```

Comment:
nit: If we do choose to just hash `id`, then we can remove `_flatten_mesh_list`.